text/label/sentencepiece/apply.py

```diff
@@ -0,0 +1,104 @@
+        :param map_unk: when encoding string to string, spm won't map oov symbol to <unk> but keep it as is.
+            This option forces the oov labels to be <unk> by cecking encoded indices.
+        :param gzip_output: use gzip on the output text
+        :param mini_task: if the Job should run locally, e.g. only a small (<1M lines) text should be processed
+        """
+        self.text_file = text_file
+        self.sentencepiece_model = sentencepiece_model
+        self.map_unk = map_unk
+        self.gzip_output = gzip_output
+        self.buffer_size = buffer_size
+
+        self.out_sentencepiece_text = self.output_path(
+            "words_to_sentencepiece.txt.gz" if gzip_output else "words_to_sentencepiece.txt"
+        )
+
+        self.mini_task = mini_task
+        self.rqmt = {"cpu": 1, "mem": 2, "time": 2}
+
+    def tasks(self):
+        if self.mini_task:
+            yield Task("run", mini_task=True)
+        else:
+            yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        import sentencepiece
+
+        spm = sentencepiece.SentencePieceProcessor(model_file=self.sentencepiece_model.get_path())
+        unk_id = spm.unk_id()
+        with tempfile.TemporaryDirectory(prefix=gs.TMP_PREFIX) as tmp:
+            input_file = self.text_file.get_path()
+            tmp_infile = os.path.join(tmp, "in_text.txt")
+            tmp_outfile = os.path.join(tmp, "out_text.txt")
+            # normalize text format
+            with util.uopen(tmp_infile, "wt") as out:
+                sp.call(["zcat", "-f", input_file], stdout=out)
+
+            with util.uopen(tmp_infile, "rt") as fin, util.uopen(tmp_outfile, "wt") as fout:
+                buf = []
+                for lineno, line in enumerate(fin, 1):
+                    pieces = spm.encode(line.rstrip("\n"), out_type=str)
+                    if self.map_unk:
+                        pieces_id = spm.encode(line.rstrip("\n"))
+                        assert len(pieces_id) == len(pieces)
+                        if unk_id in pieces_id:
+                            pieces[:] = ["<unk>" if x == unk_id else y for x, y in zip(pieces_id, pieces)]
+                    buf.append(" ".join(pieces) + "\n")
+                    # Flush periodically to avoid large memory spikes
+                    if lineno % self.buffer_size == 0:
+                        fout.writelines(buf)
+                        buf.clear()
+                if buf:  # write any remainder
+                    fout.writelines(buf)
+
+            if self.gzip_output:
+                with util.uopen(tmp_outfile, "rt") as fin, util.uopen(self.out_sentencepiece_text, "wb") as fout:
+                    sp.call(["gzip"], stdin=fin, stdout=fout)
+            else:
+                shutil.copy(tmp_outfile, self.out_sentencepiece_text.get_path())
+
+    @classmethod
+    def hash(cls, parsed_args):
+        del parsed_args["mini_task"]
```

Comment:
You should also exclude `buffer_size`.

```suggestion
        del parsed_args["mini_task"]
        del parsed_args["buffer_size"]
```